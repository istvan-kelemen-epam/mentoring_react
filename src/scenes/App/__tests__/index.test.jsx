import React from 'react';
import { App } from '../index.js';
import renderer from 'react-test-renderer';

describe('scenes/App', () => {
	it('should create snapshot', () => {
		const tree = renderer.create(<App />);
		expect(tree).toMatchSnapshot();
	});

	describe('.updateSearchExpression', () => {
		let app;

		beforeEach(() => {
			app = new App();
			app.setState = jest.fn();
		});

		it('should not update state', () => {
			const searchExpression = '';

			app.updateSearchExpression(searchExpression);

			expect(app.setState).toHaveBeenCalledWith({
				searchExpression: searchExpression
			});
		});

		it('should update state', () => {
			const searchExpression = 'search expression';

			app.updateSearchExpression(searchExpression);

			expect(app.setState).toHaveBeenCalledWith({
				searchExpression: searchExpression
			});
		});

		it('should search by title', () => {
			const searchExpression = 'search expression';

			app.state = {
				searchBy: App.SEARCH_BY.TITLE
			};
			app.updateSearchExpression(searchExpression);

			expect(app.setState).toHaveBeenCalledWith({
				searchExpression: searchExpression
			});
		});

		it('should search by genre', () => {
			const searchExpression = 'search expression';

			app.state = {
				searchBy: App.SEARCH_BY.GENRE
			};
			app.updateSearchExpression(searchExpression);

			expect(app.setState).toHaveBeenCalledWith({
				searchExpression: searchExpression
			});
		});
	});

	describe('.updateSearchBy', () => {
		it('should update state', () => {
			const app = new App();
			const searchBy = App.SEARCH_BY.TITLE;

			app.setState = jest.fn();

			app.updateSearchBy(searchBy);

			expect(app.setState).toHaveBeenCalledWith({
				searchBy: searchBy
			});
		});
	});
});