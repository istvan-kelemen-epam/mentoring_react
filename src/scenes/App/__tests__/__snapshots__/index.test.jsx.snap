// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scenes/App should create snapshot 1`] = `
<main>
  <section
    className="main-header"
  >
    <h1
      className="main-caption"
    >
      netflixroulette
    </h1>
    <section
      className="search"
    >
      <h2>
        Find your movie
      </h2>
      <input
        placeholder="Search expression"
        type="text"
      />
      <section
        className="search-by"
      >
        <header>
          Search by
        </header>
        <main>
          <button
            className="selected"
            onClick={[Function]}
          >
            Title
          </button>
          <button
            className=""
            onClick={[Function]}
          >
            Genre
          </button>
        </main>
      </section>
      <section
        className="search-button"
      >
        <button
          onClick={[Function]}
        >
          Search
        </button>
      </section>
    </section>
  </section>
  <div>
    
  </div>
  <section
    className="status"
  >
    <span
      className="status__found"
    >
      
    </span>
    <span
      className="status__sort-by"
    >
      <span
        className="status__sort-by-caption"
      >
        Sort by
      </span>
      <span
        className="status__sort-by-value"
      >
        release date
      </span>
    </span>
    <span
      className="status__rating"
    >
      rating
    </span>
  </section>
  <section
    className="result"
  />
</main>
`;
